src/components/CustomCollection/CustomCollection.tsx

```diff
@@ -48,13 +46,29 @@ const CustomCollection = ({
   const urlInputValue = useRef<string>('')
   const addCustomLinkModal = useRef<ModalRef>(null)
   const linkInput = useRef<ComboBoxRef>(null)
+  const [isEditing, setEditing] = useState(false)
```

Comment:
maybe we should make these states a little more explicit now that there's a few things going on in this component? like, `isEditingTitle` and `isAddingLink`